core/server/src/main/java/tachyon/web/WebInterfaceMasterMetricsServlet.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.web;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.TreeMap;
+
+import javax.annotation.concurrent.ThreadSafe;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.codahale.metrics.Counter;
+import com.codahale.metrics.Metric;
+import com.codahale.metrics.MetricFilter;
+import com.codahale.metrics.MetricRegistry;
+
+import com.google.common.base.Preconditions;
+
+import tachyon.master.MasterContext;
+import tachyon.master.MasterSource;
+import tachyon.master.TachyonMaster;
+import tachyon.util.CommonUtils;
+
+/**
+ * Servlet that provides data for viewing the master metrics values
+ */
+@ThreadSafe
+public final class WebInterfaceMasterMetricsServlet extends WebInterfaceAbstractMetricsServlet {
+
+  private static final long serialVersionUID = -1481253168100363787L;
+  private final transient TachyonMaster mMaster;
+
+  /**
+   * Creates a new instance of {@link WebInterfaceMasterMetricsServlet}.
+   *
+   * @param master Tachyon master
+   */
+  public WebInterfaceMasterMetricsServlet(TachyonMaster master) {
```

Comment:
only pass the Metrics System into the constructor... the servlet should not need access to anything else